packages/render/src/render-async.ts

```diff
@@ -1,34 +1,46 @@
-/* eslint-disable @typescript-eslint/no-unnecessary-condition */
 import { convert } from "html-to-text";
-import type { ReactDOMServerReadableStream } from "react-dom/server";
+import type {
+  PipeableStream,
+  ReactDOMServerReadableStream,
+} from "react-dom/server";
 import { pretty } from "./utils/pretty";
 import { plainTextSelectors } from "./plain-text-selectors";
 import type { Options } from "./options";
 
 const decoder = new TextDecoder("utf-8");
 
 const readStream = async (
-  readableStream: NodeJS.ReadableStream | ReactDOMServerReadableStream,
+  stream: PipeableStream | ReactDOMServerReadableStream,
 ) => {
   let result = "";
 
-  if ("allReady" in readableStream) {
-    const reader = readableStream.getReader();
-
-    // eslint-disable-next-line no-constant-condition
-    while (true) {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, no-await-in-loop
-      const { value, done } = await reader.read();
-      if (done) {
-        break;
-      }
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      result += decoder.decode(value);
-    }
+  if ("pipeTo" in stream) {
+    // means it's a readable stream
+    const writableStream = new WritableStream({
+      write(chunk: BufferSource) {
+        result += decoder.decode(chunk);
+      },
+    });
+    await stream.pipeTo(writableStream);
   } else {
-    for await (const chunk of readableStream) {
-      result += decoder.decode(Buffer.from(chunk));
-    }
+    const {
+      default: { Writable },
+    } = await import("node:stream");
```

Comment:
When this gets compiled to a require statement by the build, I get the following in my native ESM project:

```
Error: Dynamic require of "stream" is not supported
```

All server-side JavaScript environment support dynamic imports. Could we not transpile the dynamic import?